coremltools/converters/nnssa/coreml/graph_pass/mlmodel_passes.py

```diff
@@ -244,27 +245,49 @@ def remove_redundant_transposes(spec):
     that compose to the identity.
     """
 
+    def _get_input_output_hashmap(nn_layers):
+        # Hashmap for layers
+        output_to_layers = {}
+        for layer in nn_layers:
+            for input in layer.input:
+                if not input in output_to_layers:
+                    output_to_layers[input] = [layer]
+                else:
+                    output_to_layers[input].append(layer)
+
+        input_to_layer = {}
+        for layer in nn_layers:
+            for output in layer.output:
+                assert(output not in input_to_layer)
+                input_to_layer[output] = layer
```

Comment:
similarly,
`input_to_layers` : [str:layer_proto_message]: blob name: parent layers

again I think something like `blob_name_to_parent_layer` is a better term.

Reply:
good!